src/SourceLocator/Type/AggregateSourceLocator.php

```diff
@@ -32,14 +30,10 @@ public function locateIdentifier(Reflector $reflector, Identifier $identifier):
         return null;
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    public function locateIdentifiersByType(Reflector $reflector, IdentifierType $identifierType): array
+    public function locateIdentifiersByType(Reflector $reflector, IdentifierType $identifierType): Generator
```

Comment:
The implementation looks good, but the return type should probably be `iterable`